rfc/20240529-OpenTofu-Specific-Code-Override.md

```diff
@@ -0,0 +1,161 @@
+In case we want to create a new plugin, we might want to consider that on 2023, Jetbrains started [supporting LSP for Plugin Developers](https://blog.jetbrains.com/platform/2023/07/lsp-for-plugin-developers/) and we can use the [OpenTofu Language Server](https://github.com/gamunu/opentofu-ls) for the development of the new plugin. Although, it can only support paid IntelliJ-based IDEs, as explained [here](https://blog.jetbrains.com/platform/2023/07/lsp-for-plugin-developers/).
+
+#### Configurable plugin to support `.tf` and `.tofu` files based on user preference
+In both Jetbrains IDEs and VSCode, plugins can be configured by the end user in the Settings menu. Possibly, we can add an optional setting to the plugin that allows it to recognize `.tf` files as OpenTofu files. This option can be very helpful for users switching to OpenTofu from Terraform, and want to get quick support for OpenTofu in the IDE without changing the file extension to `.tofu`. On the other hands, people who want to use both Terraform and OpenTofu in the same project can turn this setting off, and the plugin will not go over the `.tf` files. This setting should be per project and not for the whole IDE (which should be possible based on [Jetbrains documentation](https://www.jetbrains.com/help/idea/configuring-project-and-ide-settings.html) and [VSCode documentation](https://code.visualstudio.com/docs/getstarted/settings)).
+
+#### LSP and Other IDEs
+The [OpenTofu Language Server](https://github.com/gamunu/opentofu-ls) can be used with other IDEs that support the [Language Server Protocol](https://en.wikipedia.org/wiki/Language_Server_Protocol).
+There are different websites that provide a list of LSP implementations (Like, [this list by Microsoft](https://microsoft.github.io/language-server-protocol/implementors/servers/) and [Langserver.org](https://langserver.org/)) and we should enlist the OpenTofu Language Server, so users will be able to easily find it.
+
+### 3rd Party Tools
+Today, some 3rd party tools officially support OpenTofu. Many of them are listed in the [awesome-opentofu repository](https://github.com/virtualroot/awesome-opentofu). Implementing the solution suggested in this RFC might affect 3rd party tools maintainers and they will have to create adjustment support the new `.tofu` extension files. As part of this RFC we should map those tools, understand if they are impacted by this change, and reach out to the maintainers and notify them about this change or contribute.
+
+TODO: add a list of 3rd party tools that are affected by this change.
+
+Furthermore, this change could potentially incentivize other tools to also support OpenTofu. With a clear strategy for managing code divergence in place, maintainers of those tools may gain a better understanding of how to provide support for OpenTofu. Additionally, they can begin by accommodating the new file extensions that we will be introducing in this RFC.
+
+## Technical Approach
```

Comment:
I would also like to add that we should log very verbosely to the debug logs about what we're overriding and why. To help developers understand any issues they hit when using .tofu files.

Reply:
Good idea! added it [here](https://github.com/opentofu/opentofu/pull/1699/commits/a8980723f0e507ce38b160c6d8c77e8018df7624)